### DISCUSSION TITLE
Run 'make check' only when bottling

### DISCUSSION THREAD

**Discussion by mahrud** [Homebrew/discussions#5466] [Tap maintenance and Homebrew development]
I want to add `make check` to a few formulae on a self-maintained tap, but I also want the ability to skip the checks occasionally (or vice versa, only run `make check` occasionally).

I thought an optional argument would be perfect for this purpose, but that doesn't seem to be supported anymore, or at least I can't find any reference to how to add one, so if this is still possible through a secret syntax let me know!

Otherwise, is there a way to run a command in the 'install' section of the formula only if brew is building a bottle, as opposed to building from source without bottling? This way GitHub builds will still run the checks, but if I ever want to build a new version from source I don't need to wait for the tests to pass (and more importantly, if a test fails the whole build isn't aborted).

**Comment by ZhongRuoyu** [comment]
You can guard the `make check` call with `if build.bottle?`; in that way it will only be invoked when building a bottle. Sample usage: https://github.com/Homebrew/homebrew-core/blob/b254c9f8b8dd578afece13915c0b4adab7930f35/Formula/n/node.rb#L96
